sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -130,6 +125,19 @@ def test_zero_power():
     raises(ValueError, lambda:MatPow(z2, -1).doit())
 
 
+def test_OneMatrix_power():
+    o = OneMatrix(3, 3)
+    assert o ** 0 == Identity(3)
+    assert o ** 1 == o
+    assert o * o == o ** 2 == 3 * o
+    assert o * o * o == o ** 3 == 9 * o
+
+    o = OneMatrix(n, n)
+    assert o * o == o ** 2 == n * o
+    # simplify currently necessary as n ** (n - 2) * n does not produce n ** (n - 1)
+    assert simplify(o ** (n - 1) * o) == o ** n == n ** (n - 1) * o
```

Comment:
It doesn't really matter here but generally we prefer to avoid using full `simplify` in library or testing code as it uses heuristics and can be slow.

It is better (especially if you know exactly what the expression is) to use the particular kind of simplification that you know is necessary. For example in this case what you want is `powsimp`:
```julia
In [10]: powsimp(n**(n-2)*n)                                                                                                      
Out[10]: 
 n - 1
n 
```
Full `simplify` will call `powsimp` (perhaps several times) and do a whole load of other stuff so if you just want `powsimp` it's best to do that directly.

Reply:
Good point, thanks.